File: kattis/social_distancing.py
# https://open.kattis.com/problems/socialdistancing2
# 1s
# 2.2 Easy

# 128 chars: Top 10 Place 3 (shortest Python3 solution!). Thomas Feld has a Ruby solution of 60 chars!
n=int(input().split()[0])
s=list(map(int,input().split()));s+=[s[0]+n]
print(sum([(s[i+1]-s[i]-2)//2 for i in range(len(s)-1)]))

File: kattis/social_distancing.pl
% https://open.kattis.com/problems/socialdistancing2
% 1s
% 2.2 Easy

% 176 chars: Top 10 place 6
% Here's the original logic: The two "-1" are reduced to -2 for brevity.
% findall(C,(nextto(A,B,L), D is B-A-1,C is(D-1)div 2),Ds),

% Shorter: 169 chars: Top 10, still place 6
main :-
    readln([N,_,H|T],end_of_file),
    Y is H+N,
    append([H|T],[Y],L),
    findall(D,(nextto(A,B,L),D is(B-A-2)div 2),Ds),
    sum_list(Ds,R),
    writeln(R).

/*
% Compressed: 139 chars: Top 10 place 4 (before my Python3 solution)
main:-readln([N,_,H|T],end_of_file),Y is H+N,append([H|T],[Y],L),
findall(D,(nextto(A,B,L),D is(B-A-2)div 2),Ds),sum_list(Ds,R),writeln(R).
*/